src/Query/Grouper.ts

```diff
@@ -36,12 +37,24 @@ export class Grouper {
 
     /**
      * Whether the headings for this group should be reversed.
+     * TODO now reverse used only in TaskGroups.toString(), shall be removed.
      */
     public readonly reverse: boolean;
 
-    constructor(property: string, grouper: GrouperFunction, reverse: boolean) {
+    public readonly comparator: Comparator;
+
+    constructor(property: string, grouper: GrouperFunction, reverse: boolean, comparator: Comparator) {
         this.property = property;
         this.grouper = grouper;
         this.reverse = reverse;
+        this.comparator = (a: Task, b: Task) => {
+            const result = comparator(a, b);
```

Comment:
Maybe I'm missing something but we talked elsewhere about comparator() throwing if sorting is unimplemented in the field...

Reply:
True, but this is just refactoring - moving the comparator from `TaskGroups.sortTaskGroups` into `Field.defaultComparator`. For now `Field.comparator()` is never called.

Reply:
Ah - right. Now I see what's going on.

I need to actually test this out and think about it some more.

I would be a lot more comfortable, and the review would be a lot quicker, if there were already the thorough group-sorting tests that we have talked about previously.

Reply:
If I recall correctly, you said we don't need to have extensive tests till the end to start changing the code. I may have misunderstood.

Reply:
I can quite believe I said that, but it wasn't my intention.

Once it became possible to change the sort order of groups, it revealed a gaping hole in the grouping tests, which had previously only tested the names of groups and not the order.

I would like that hole to be fixed for all fields before the grouping code is changed please.